### PR TITLE
`minikube addons list` tests

### PR DESCRIPTION
**Pull request by spowelljr**
Follow up to https://github.com/kubernetes/minikube/pull/12837

Adding tests and fix JSON output.

### CODE REVIEW

File: cmd/minikube/cmd/config/addons_list.go [hunk 1]
```diff
@@ -137,6 +137,10 @@ var printAddonsList = func(cc *config.ClusterConfig) {
 }
 
 var printAddonsJSON = func(cc *config.ClusterConfig) {
+	if cc == nil {
```
Comment (sharifelgamal) [inline cmd/minikube/cmd/config/addons_list.go:140]: so for JSON we're just returning nothing if there's no cluster?

Comment (sharifelgamal) [reply cmd/minikube/cmd/config/addons_list.go:140]: could we not do the same thing we do for table mode? like output the list of addons without a status?

Comment (spowelljr) [reply cmd/minikube/cmd/config/addons_list.go]: Yup, done

**Review by sharifelgamal — APPROVED** [review]

**Pull request merged**